node/pkg/nodeinit/calico-init_linux.go

```diff
@@ -50,6 +62,156 @@ func Run(bestEffort bool) {
 			os.Exit(3)
 		}
 	}
+	serviceAddr := os.Getenv("KUBERNETES_SERVICE_HOST_PORT")
+	endpointAddrs := os.Getenv("KUBERNETES_APISERVER_ENDPOINTS")
+	if serviceAddr != "" && endpointAddrs != "" {
+		_, err = initBPFNetwork(serviceAddr, endpointAddrs)
+		if err != nil {
+			logrus.WithError(err).Error("Failed to initialize BPF network.")
+			if !bestEffort {
+				os.Exit(4)
+			}
+		}
+	}
+}
+
+func initBPFNetwork(serviceAddr, endpointAddrs string) (*bpfmap.Maps, error) {
+	logrus.Info("Initializing BPF network.")
+
+	servicesIPPort, err := parseCommaSeparatedIPPorts(serviceAddr)
+	if err != nil {
+		return nil, err
+	}
+	endpointsIPPort, err := parseCommaSeparatedIPPorts(endpointAddrs)
+	if err != nil {
+		return nil, err
+	}
+
+	countIPv4, countIPv6 := countIPv4AndIPv6(append(endpointsIPPort, servicesIPPort...))
+
+	bpfMaps, err := bpfmap.CreateBPFMaps(countIPv6 > 0)
+	if err != nil {
+		logrus.WithError(err).Error("Failed to create bpf maps.")
+		return nil, err
+	}
+
+	id := uint32(0)
+	for _, service := range servicesIPPort {
+		if service.IsIPv4 {
+			err = bpfMaps.V4.FrontendMap.Update(
+				bpfnat.NewNATKey(service.IP, service.Port, uint8(layers.IPProtocolTCP)).AsBytes(),
+				bpfnat.NewNATValue(id, uint32(countIPv4), 0, 0).AsBytes(),
+			)
+			if err != nil {
+				logrus.WithError(err).Error("Failed to add IPv4 set entry in the frontend map.")
+				return nil, err
+			}
+			// Add the endpoints to the backend map.
+			for index, endpoint := range endpointsIPPort {
```

Comment:
why not to flip this, first write the backends, count them and then write the front end.